### ISSUE TITLE
undefined's All-Time Stats

### ISSUE THREAD

**Issue by Saibamen** [timiimit/UT4MasterServer#109]
Related: #22 
FYI: @belmirp 

![image](https://user-images.githubusercontent.com/905878/213925415-4b2f9a25-ced4-4548-bb5d-c101eb4c6724.png)

### Repro steps

1. Create new account
2. Login into game, **don't play**
3. Click on "Show stats for this player, friends, and recent opponents." button
![image](https://user-images.githubusercontent.com/905878/213925485-0823adc6-94f1-457e-a688-bea573f7e7df.png)

**Comment by belmirp** [comment]
@Saibamen This is on the stats page, but player's name is fetched from other endpoint I believe. Stats endpoint is only returning weapon statistics in an array. Generally, this page triggers couple of HTTP calls to different endpoints: players list, MMR ratings, recent matches and the weapon stats (which is issue #22).

**Comment by Saibamen** [comment]
After playing one match (or training), I can see my username:
![image](https://user-images.githubusercontent.com/905878/213926558-bb98fe5c-30fb-43f3-b2b0-de5fa1676973.png)

**Comment by belmirp** [comment]
That has to be related to something else, since I am not returning in any case player's name. I will check what the page is calling.

**Comment by timiimit** [comment]
It doesn't seem like any endpoint which could get username is being called when stats open. maybe its retrieved from stats.json somehow.

**Comment by belmirp** [comment]
Yeah, it seems it is extracted from stats.json which is the last 5 matches.
This is the code in stats.html that populates it:
`document.getElementById('PlayerName').innerHTML = Stats.PlayerName;`

**Comment by timiimit** [comment]
Then the only way to "fix" this would be to respond with a fake stats.json when its empty. I'll try to do that.
